### PR TITLE
Cache lookups for EventUser on event creation

### PR DESCRIPTION
**Pull request by dcramer**
This ensures that low cardinality (or high throughput) scenarios defer most IntegrityError scenarios, which can become very common.

### CODE REVIEW

File: src/sentry/event_manager.py [hunk 1]
```diff
@@ -542,11 +543,15 @@ def _get_event_user(self, project, data):
...
-            pass
+        cache_key = 'euser:{}'.format(md5(euser.tag_value).hexdigest())
+        euser = cache.get(cache_key)
+        if euser is None:
+            try:
+                with transaction.atomic():
+                    euser.save()
+            except IntegrityError:
+                pass
+            cache.set(euser, 300)
```
Comment (mattrobenolt) [inline src/sentry/event_manager.py]: Missing the cache_key.

File: src/sentry/event_manager.py [hunk 2]
```diff
@@ -542,11 +543,15 @@ def _get_event_user(self, project, data):
         if not euser.tag_value:
             return
 
-        try:
-            with transaction.atomic():
-                euser.save()
-        except IntegrityError:
-            pass
+        cache_key = 'euser:{}'.format(md5(euser.tag_value).hexdigest())
```
Comment (mattrobenolt) [inline src/sentry/event_manager.py]: Wouldn't we want this to also include project id?

Comment (mattrobenolt) [inline src/sentry/event_manager.py]: I assume it should match the unique_together index or we're going to have issues here from my understanding.

Comment (dcramer) [inline src/sentry/event_manager.py]: mm yeah we would

**Comment by mattrobenolt** [comment]
:+1:

**Pull request merged**